server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -49,21 +50,42 @@ protected boolean searchWithCollector(
         boolean hasFilterCollector,
         boolean hasTimeout
     ) throws IOException {
-        return searchWithCollectorManager(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
+        // create the top docs collector
+        final TopDocsCollectorContext topDocsFactory = createTopDocsCollectorContext(searchContext, hasFilterCollector);
+        return searchWithCollector(searchContext, searcher, query, collectors, topDocsFactory, hasFilterCollector, hasTimeout);
+    }
+
+    protected boolean searchWithCollector(
```

Comment:
I think there is an `@Override` missing here. Also can we just call `searchWithCollectorManager` directly on L55 above?

Reply:
Will add @Override in commit. We have to call new overloaded method, this allows to child implementation to override and pass empty context if needed. That was the main point for this change